src/MapUtil/MapUtil.js

```diff
@@ -393,6 +404,158 @@ export class MapUtil {
+        attribution: attributionString
+      };
+      return /** @type {import("../types").InkmapWmtsLayer} */ (wmtsLayer);
+    } else if (source instanceof OlSourceOSM) {
+      const osmLayer = {
+        type: 'XYZ',
+        url: 'https://{a-c}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+        opacity: opacity,
+        attribution: '© OpenStreetMap (www.openstreetmap.org)',
+        tiled: true
+      };
+      return /** @type {import("../types").InkmapOsmLayer} */ (osmLayer);
+    } else if (source instanceof OlSourceVector) {
+      const geojson = new OlFormatGeoJSON().writeFeaturesObject(source.getFeatures());
+      const parser = new OpenLayersParser();
+      const geojsonLayerConfig = {
+        type: 'GeoJSON',
+        geojson: geojson,
+        attribution: attributionString,
+        style: undefined
+      };
+
+      let olStyle = null;
+
+      if (olLayer instanceof OlLayerVector) {
+        olStyle = olLayer.getStyle();
+      }
+
+      // todo: support stylefunction / different styles per feature
+      // const styles = source.getFeatures()?.map(f => f.getStyle());
+
+      if (olStyle) {
+        const gsStyle = await parser.readStyle(olStyle);
+        if (gsStyle.errors) {
+          Logger.error('Geostyler errors: ', gsStyle.errors);
+        }
+        if (gsStyle.warnings) {
+          Logger.warn('Geostyler warnings: ', gsStyle.warnings);
+        }
+        if (gsStyle.unsupportedProperties) {
+          Logger.warn('Detected unsupported style properties: ', gsStyle.unsupportedProperties);
+        }
+        if (gsStyle.output) {
+          // @ts-ignore
+          geojsonLayerConfig.style = gsStyle.output;
+        }
+      }
+      return /** @type {import("../types").InkmapGeoJsonLayer} */ (geojsonLayerConfig);
+    }
+    return null;
+  }
+
+  /**
+   * Converts a given OpenLayers map to an inkmap spec. Only returns options which can be
+   * derived from a map (center, scale, projection, layers).
+   *
+   * @param {import("ol/Map").default} olMap The ol map.
+   *
+   * @return {Promise<Partial<import("../types").InkmapPrintSpec>>} Promise of the inmkap print spec.
+   */
+  static async generatePrintConfig(olMap) {
+    const unit = olMap.getView().getProjection().getUnits();
+    const resolution = olMap.getView().getResolution();
+    const projection =  olMap.getView().getProjection().getCode();
+    if (resolution === undefined) {
+      throw new Error('Can not determine resolution from map');
+    }
+
+    const scale = MapUtil.getScaleForResolution(resolution, unit);
+    const center = olMap?.getView().getCenter();
+    if (!unit || !center || !_isFinite(resolution)) {
+      throw new Error('Can not determine unit / resolution from map');
+    }
+    const centerLonLat = toLonLat(center, projection);
+
+    const layerPromises = olMap.getAllLayers()
+      .map(MapUtil.mapOlLayerToInkmap);
+
+    // @ts-ignore
+    return Promise.all(layerPromises)
+      .then((responses) => {
```

Comment:
Can we make use of `async/await` here instead?

Reply:
yes, done in 9173feb850da131ae2c6ba26e8ebf72f7b79383d